src/libs/SidebarUtils.js

```diff
@@ -85,22 +101,43 @@ function setIsSidebarLoadedReady() {
  * @returns {String[]} An array of reportIDs sorted in the proper order
  */
 function getOrderedReportIDs(currentReportId, allReportsDict, betas, policies, priorityMode, allReportActions) {
+    // Generate a unique cache key based on the function arguments
+    const cachedReportsKey = JSON.stringify(
+        // eslint-disable-next-line es/no-optional-chaining
+        [currentReportId, allReportsDict, betas, policies, priorityMode, allReportActions[`reportActions_${currentReportId}`]?.length || 1],
+        (key, value) => {
+            // Exclude 'participantAccountIDs', 'participants' and 'lastMessageText' properties from all objects in the 'allReportsDict' array
+            if (key === 'participantAccountIDs' || key === 'participants' || key === 'lastMessageText') {
+                return undefined;
+            }
+            return value;
+        },
+    );
+
+    // // Check if the result is already in the cache
+    if (reportIDsCache.has(cachedReportsKey) && hasInitialReportActions) {
+        return reportIDsCache.get(cachedReportsKey);
+    }
+    hasInitialReportActions = Object.values(lastReportActions).length > 0;
```

Comment:
We need this flag to update it after "if", not just to verify inside.
This is needed to prevent an issue with Onyx when its empty for a second render in ReportActionsUtils function.